ICSharpCode.Decompiler/CSharp/CSharpDecompiler.cs

```diff
@@ -106,6 +106,7 @@ public static List<IILTransform> GetILTransforms()
 				new SwitchDetection(),
 				new SwitchOnStringTransform(),
 				new SwitchOnNullableTransform(),
+				new SwitchExpressionDefaultCaseTransform(),
```

Comment:
I think that doing this as a post-processing transform of switch detection is a problem. At this point there is no separate block container for `switch` (`BlockContainerKind.Switch` is not yet set)... this transformation only happens later in `LoopDetection`. So this transform would have to perform all the validation that happens in the other `switch` detection transforms again.